Immense.RemoteControl.Server/wwwroot/src/CaptureProcessor.ts

```diff
@@ -1,45 +1,65 @@
 ﻿import { ViewerApp } from "./App.js";
-import { ScreenCaptureDto } from "./Interfaces/Dtos.js";
+import { StreamingState } from "./Models/StreamingState.js";
 import { Screen2DContext } from "./UI.js";
+import { GetUint64 } from "./Utilities.js";
 
-const Partials: Record<string, Array<Uint8Array>> = {};
+const FrameHeaderSize = 28;
 
-export async function HandleCaptureReceived(screenCapture: ScreenCaptureDto) {
 
-    if (!Partials[screenCapture.InstanceId]) {
-        Partials[screenCapture.InstanceId] = [];
+export async function ProcessStream(streamingState: StreamingState): Promise<void> {
+    if (streamingState.StreamEnded) {
+        return;
     }
 
-    Partials[screenCapture.InstanceId].push(screenCapture.ImageBytes);
+    try {
+        const chunks = streamingState.ReceivedChunks.splice(0);
+        streamingState.Buffer = new Blob([streamingState.Buffer, ...chunks]);
 
-    if (!screenCapture.IsLastChunk) {
-        return;
-    }
+        const bufferSize = streamingState.Buffer.size;
+
+        if (bufferSize < FrameHeaderSize) {
+            return;
+        }
+
+        const headerBlob = streamingState.Buffer.slice(0, FrameHeaderSize);
+        const buffer = await headerBlob.arrayBuffer();
```

Comment:
Not related/suggestion, but if this ever becomes Blazor/WASM, be really cool to use the [System.Threading.Channels](https://learn.microsoft.com/en-us/dotnet/core/extensions/channels) which supports back pressure and all that for you. We use it in the Ephemeral and it works well imo.

Reply:
Yeah, that would be very much preferable.  Cross fingers!  :D